src/parser/ion_1_0/binary.rs

```diff
@@ -422,18 +423,42 @@ fn parse_decimal(typed_value: TypedValue) -> ParseResult<&[u8], Option<Decimal>>
 /// text encoding are in the local time! This means that transcoding requires a conversion between
 /// UTC and local time.
 fn parse_timestamp(typed_value: TypedValue) -> ParseResult<&[u8], Option<Timestamp>> {
+    fn time_error(i: &[u8], component: TimeComponent, value: BigInt) -> nom::Err<IonError<&[u8]>> {
+        Err::Failure(IonError::from_format_error(
+            i,
+            FormatError::Binary(BinaryFormatError::TimeComponentRange(component, value)),
+        ))
+    }
+
     match typed_value.length_code {
         LengthCode::L15 => Ok(None),
         _ => {
             let (rest, offset) = take_var_int(typed_value.rep)?;
+            let offset = offset
+                .to_i32()
+                .ok_or_else(|| time_error(typed_value.index, TimeComponent::Offset, offset))?;
```

Comment:
All throughout where this pattern shows up in this file I'd prefer a match with a return branch over an `ok_or_else` with a closure containing an error, and a `?`.

Reply:
Will fix in a couple minutes.